pkg/controllers/netpol/pod.go

```diff
@@ -53,12 +105,64 @@ func (npc *NetworkPolicyController) handlePodDelete(obj interface{}) {
 			return
 		}
 	}
-	glog.V(2).Infof("Received pod: %s/%s delete event", pod.Namespace, pod.Name)
+	glog.V(2).Infof("Received pod:%s/%s delete event", pod.Namespace, pod.Name)
+
+	// skip processing update to pods in host network
+	if pod.Spec.HostNetwork {
+		return
+	}
+
+	podInfo := podInfo{ip: pod.Status.PodIP,
+		name:      pod.ObjectMeta.Name,
+		namespace: pod.ObjectMeta.Namespace,
+		labels:    pod.ObjectMeta.Labels}
+
+	err := npc.syncAffectedNetworkPolicyChains(&podInfo, syncVersion)
+	if err != nil {
+		glog.Errorf("failed to refresh network policy chains affected by pod %s/%s delete event due to %s", pod.Namespace, pod.Name, err.Error())
+	}
+
+	// cleanup of firewall chains needed only for local pods
+	if !isLocalPod(pod, npc.nodeIP.String()) {
+		return
+	}
+
+	npc.mu.Lock()
```

Comment:
How come you don't grab the lock here before you call `npc.syncAffectedNetworkPolicyChains` above? You hold the lock before that statement in `processPodAddUpdateEvents()`